spec/models/portfolio_spec.rb

```diff
@@ -18,6 +18,35 @@
     end
   end
 
+  context "model fails validation on unique name by same tenant" do
+    let(:portfolio_copy) { create(:portfolio, :without_tenant) }
+    let(:second_tenant)  { create(:tenant) }
+
+    before do
+      ActsAsTenant.current_tenant = tenant
+    end
+
+    it "will fail validation if tenants create portfolios with the same name" do
+      portfolio.update(:name => "samename")
+      portfolio_copy.update(:name => "samename")
+
+      expect(portfolio).to be_valid
+      expect(portfolio_copy).to_not be_valid
+      expect(portfolio_copy.errors.messages[:name]).to_not be_nil
+
+      expect{ portfolio_copy.save! }.to raise_exception(ActiveRecord::RecordInvalid)
+    end
+
+    it "will pass validation if different tenants try to create portfolios" do
+      portfolio.update(:name => "samename")
```

Comment:
Same point here, the method name talks about a different tenant in a context that declares all the same tenant.